File: frontend/src/core/models/task/task-custom-field.ts
import { CheckboxField } from './checkbox-field';
import { DropdownField } from './dropdown-field';
import { LabelField } from './label-field';
import { TaskFieldType } from './task-field-types';

export interface TaskCustomField {
    name: string,
    type: TaskFieldType,
    fieldId?: string,

    labels?: LabelField[],
    dropdown?: DropdownField,
    checkboxes?: CheckboxField[],
}

File: backend/Tasque.Core.Common/Entities/Task/TaskCustomField.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Tasque.Core.Common.Entities.Task
{
    public class TaskCustomField
    {

        public string Name { get; set; }

        public int Type { get; set; }

        public List<LabelField>? Labels { get; set; }

        public DropdownField? Dropdown { get; set; }
    }
}
